builder/amazon/common/step_source_ami_info.go

```diff
@@ -53,44 +52,13 @@ func (s *StepSourceAMIInfo) Run(ctx context.Context, state multistep.StateBag) m
 		params.ImageIds = []*string{&s.SourceAmi}
 	}
 
-	// We have filters to apply
-	if len(s.AmiFilters.Filters) > 0 {
-		params.Filters = buildEc2Filters(s.AmiFilters.Filters)
-	}
-	if len(s.AmiFilters.Owners) > 0 {
-		params.Owners = s.AmiFilters.GetOwners()
-	}
-
-	log.Printf("Using AMI Filters %v", params)
-	imageResp, err := ec2conn.DescribeImages(params)
+	image, err := s.AmiFilters.GetFilteredImage(params, ec2conn)
```

Comment:
I see the `ami_filter` code base is common so that users can continue to using the `source_ami_filter` configuration option. Is that correct? 

Do we envision deprecating this field in the future? Not a suggestion just a question :smile:

Reply:
Yes, It will still work for the current source_ami_filter.

I suggest we deprecate this together with JSON and then HCL2 templates will have to use the datasource instead. I think it makes more sense 🤔